healpy/__init__.py

```diff
@@ -107,16 +107,54 @@
 from ._hotspots import hotspots_healpy as hotspots
 from ._line_integral_convolution import line_integral_convolution
 
+from astropy.utils.decorators import deprecated
 
+
+@deprecated("1.15.0", alternative="configure_logging")
 def disable_warnings():
-    """Disable all healpy warnings messages for the current session
+    """healpy uses logging now
 
-    Warnings from individual functions can be disabled setting
-    ``verbose=False``.
-    Warnings can be re-enabled calling ``hp.enable_warnings()``.
+    See configure_logging
+    This function has no effect
     """
-    warnings.filterwarnings(action="ignore", module="healpy")
+    pass
 
 
+@deprecated("1.15.0", alternative="configure_logging")
 def enable_warnings():
-    warnings.simplefilter("always")
\ No newline at end of file
+    """healpy uses logging now
+
+    See configure_logging
+    This function has no effect
+    """
+    pass
+
+
+def configure_logging(
+    level=logging.DEBUG, log_format="%(name)s - %(levelname)s - %(message)s"
+):
+    """Configure logging
+
+    By default healpy only prints warning or error messages, to disable even
+    that, set the level to `logging.CRITICAL`.
+    Calling `configure_logging` with default arguments restores the same logging
+    messages of `healpy<=1.14.0`.
+
+    Parameters
+    ----------
+    level : int
+        Logging level, DEBUG, INFO, WARNING, ERROR, CRITICAL
+    log_format : str
+        Logging format string, see the logging module documentation
+    """
+    log = logging.getLogger("healpy")
+    log.setLevel(level)
+
+    # create console handler
+    handler = logging.StreamHandler()
+
+    # create formatter
+    formatter = logging.Formatter(log_format)
+    handler.setFormatter(formatter)
+
+    log.addHandler(handler)
```

Comment:
@lpsinger it is not straightforward to configure logging, setting only the level is not enough you also need to add a handler, so better provide a function.